x-pack/plugins/ml/public/application/jobs/jobs_list/components/confirm_modals/stop_datafeeds_confirm_modal.tsx

```diff
@@ -0,0 +1,141 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect, useCallback, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiSpacer,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiText,
+} from '@elastic/eui';
+import { MlSummaryJob } from '../../../../../../common/types/anomaly_detection_jobs';
+import { isManagedJob } from '../../../jobs_utils';
+// @ts-ignore
+import { stopDatafeeds } from '../utils';
+
+type ShowFunc = (jobs: MlSummaryJob[]) => void;
+
+interface Props {
+  setShowFunction(showFunc: ShowFunc): void;
+  unsetShowFunction(): void;
+  refreshJobs(): void;
+  showStopDatafeedsFlyout(job: MlSummaryJob[]): void;
+}
+
+export const StopDatafeedsConfirmModal: FC<Props> = ({
+  setShowFunction,
+  unsetShowFunction,
+  refreshJobs,
+}) => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [isManaged, setHasManaged] = useState(true);
+  const [jobsToStop, setJobsToStop] = useState<MlSummaryJob[]>([]);
+
+  const jobIds = useMemo(() => jobsToStop.map(({ id }) => id), [jobsToStop]);
+
+  useEffect(() => {
+    if (typeof setShowFunction === 'function') {
+      setShowFunction(showModal);
+    }
+    return () => {
+      if (typeof unsetShowFunction === 'function') {
+        unsetShowFunction();
+      }
+    };
+  }, []);
+
+  const showModal = useCallback((jobs: MlSummaryJob[]) => {
+    setJobsToStop(jobs);
+
+    if (jobs.some((j) => isManagedJob(j))) {
+      setModalVisible(true);
+      setHasManaged(true);
+    }
+  }, []);
+
+  const closeModal = useCallback(() => {
+    setModalVisible(false);
+    setHasManaged(false);
+  }, []);
+
+  if (modalVisible === false) {
+    return null;
+  }
+
+  if (isManaged) {
+    const title = (
+      <FormattedMessage
+        id="xpack.ml.jobsList.stopDatafeedsModal.stopDatafeedsTitle"
+        defaultMessage="Stop datafeed for {jobsCount, plural, one {{jobId}} other {# jobs}}?"
+        values={{
+          jobsCount: jobIds.length,
+          jobId: jobIds[0],
+        }}
+      />
+    );
+
+    return (
+      <EuiModal data-test-subj="mlStopDatafeedsConfirmModal" onClose={closeModal}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          <p>
+            <>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.ml.jobsList.stopDatafeedsModal.stopManagedDatafeedsDescription"
+                  defaultMessage="{jobsCount, plural, one {This job was} other {At least one of these jobs was}} deployed as part of a module; stopping {jobsCount, plural, one {it} other {them}} might impact other parts of the product."
```

Comment:
As above, we should drop the term 'module' here, and say something like `This pre-configured job was provided by Elastic...`

Reply:
Updated here https://github.com/elastic/kibana/pull/122305/commits/427be5b2ff13c2453e9b03611547f978baa19cbc